### PR TITLE
[Hub Generated] Review request for Microsoft.DevTestLab to add version stable/2018-09-15

### PR DESCRIPTION
**Pull request by joerage**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/joerage/azure-rest-api-specs/dev-devtestlabs-Microsoft.DevTestLab-2018-09-15...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (nschonni) [inline specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Adds back the correct spelling of "Response"
```suggestion
      "description": "Response body for generating an upload URI.",
```

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (sarangan12) [inline specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Removal of existing APIs need ARM review.

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (sarangan12) [inline specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Breaking changes need ARM review

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (sarangan12) [inline specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Another Breaking change...

**Review by sarangan12 — APPROVED** [review]
Changes are simple. Description changes are ok. I am Ok with enum changes also. But removal of a definition, path and changing from required:false definitely needs ARM review and approval. Adding the appropriate label. From SDK side, I do not have any issue. But I am holding off on the merge until ARM team can review.

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (joerage) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: For query parameters, the default value for 'required' is false. So removing it does not change the behavior.

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (joerage) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: This was never intended to be published. No customers should be using this.

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (joerage) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: This was never intended to be published. No customers should be using this.

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (KrisBash) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Yeah, I don't see how this would be breaking?

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (KrisBash) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: This is problematic on two fronts:
1) Removing an API path is a breaking change. If it should not have been published, we will to confirm that there is no _actual_ usage from clients. 
2) GET of serviceRunner (by name) and PUT/DELETE are still defined. A List API is required for these. Removing this means the serviceRunner child resource is not contract compliant.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Concerns with removal of LIST serviceRunners api

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (joerage) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: This is a special kind of resource. It is created in order to leverage the MSI created along. It is PUT only from our own code. However, because ARM needs to be involved (ARM triggers the creation and deletion of the MSI), PUT/DELETE/GET needs to be public. However, we have code in our RP ensuring the CREATE call is actually coming from us. If this is really a concern, I can add the LIST back. But no one but our own RP is using it (that operation has been removed months back already).

Comment (KrisBash) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: @joerage - are the GET calls made by external clients? Does this need to be documented in swagger at all if it is for service use only?

**Review by KrisBash — APPROVED** [review]
Signing off with context that removed API path has been removed from service and is not a public opeation

File: specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json
Comment (joerage) [reply specification/devtestlabs/resource-manager/Microsoft.DevTestLab/stable/2018-09-15/DTL.json]: Good point. I will remove this from swagger doc.

**Pull request merged**
